llvm/lib/Transforms/Instrumentation/AddressSanitizer.cpp

```diff
@@ -2854,7 +3053,9 @@ bool AddressSanitizer::instrumentFunction(Function &F,
     return false;
   if (F.getLinkage() == GlobalValue::AvailableExternallyLinkage) return false;
   if (!ClDebugFunc.empty() && ClDebugFunc == F.getName()) return false;
-  if (F.getName().starts_with("__asan_")) return false;
+  if (F.getName().starts_with("__asan_") ||
+      F.getName().contains("__sycl_service_kernel__"))
```

Comment:
I think that it is better to have `__sycl_service_kernel__` to be checked as a separate `if`, because it would simplify maintenance, i.e. it would be easier to resolve conflicts with upstream changes and their probability would be reduced

Reply:
Thanks!